### ISSUE TITLE
CMSFilter returns 404 for URLs containing plus sign (+) in the path

### ISSUE THREAD

**Issue by dsolistorres** [dotCMS/core#17261]
Reported via Support: Ticket [99199](https://dotcms.zendesk.com/agent/tickets/99199)
Reproducible in dotCMS 5.1.6

When uploading a file that contains a plus sign (+) as part of its name, the URL to get the file returns a 404 error code.

### Steps to reproduce the behavior:
1. Go to 'Site -> Browser' portlet.
2. Upload a file containing a plus sign (+) as part of its name, for example: `dotcms+arch.pdf`
3. Save and publish the file asset, and log out from dotCMS backend.
4. From the browser, hit the link to get the file, using  URL encoding format to escape the plus sign:
```
http://localhost:8080/training/dotcms%2Barch.pdf
```
That works.
4. Now hit the URL without using URL encoding to escape the plus sign:
```
http://localhost:8080/training/dotcms+arch.pdf
```
That return 404 error code.

### Expected behavior
URLs containing plus sign (+) in its path part should work. For the query string part of a URL, plus sign can be replaced for a white space, but that is not the case for the path part of the URL.

### Additional context
This has been reported by customer via Support: Ticket [99199](https://dotcms.zendesk.com/agent/tickets/99199)

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/core/pull/17785/files

**Comment by fabrizzio-dotCMS** [comment]
Tested with videos, file assets, and documents.
The file is always available regardless of the + sign being part of the name.
Tested both File Link 
http://localhost:8080/dA/7502c9adc4/Sample+Video.mp4?language_id=1
And resource Link 
http://localhost:8080/Sample+Video.mp4?language_id=1

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2.4 and now we are able to access the file with both ways without problems.
